client/my-sites/promote-post-i2/components/credit-balance/index.tsx

```diff
@@ -0,0 +1,37 @@
+import { memo } from '@wordpress/element';
+import { translate, numberFormat } from 'i18n-calypso';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+
+interface Props {
+	balance?: number;
+}
+
+const CreditBalance = ( { balance }: Props ) => {
+	if ( balance === undefined || balance === 0 ) {
```

Comment:
Wonder could we swap the condition this way:
```js
if ( balance > 0 ) {
    return (
        // The balance markup
    );
}

return null;
```

Reply:
I will change the condition to also exclude the case for invalid numbers (NaN).
Regarding the change of order, I think it is easier to follow the code if we take these conditions on top (like we do with loadings). 

From my point of view, if I scroll down to the bottom of the component, I would like to see the main render of that component. But it may be just my way of thinking.